lib/auth/grpcserver.go

```diff
@@ -1650,6 +1650,69 @@ func (g *GRPCServer) DeleteRole(ctx context.Context, req *proto.DeleteRoleReques
 	return &empty.Empty{}, nil
 }
 
+// MaintainSessionPresence establishes a channel used to continuously verify the presence for a session.
+func (g *GRPCServer) MaintainSessionPresence(stream proto.AuthService_MaintainSessionPresenceServer) error {
+	ctx := stream.Context()
+	actx, err := g.authenticate(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for {
+		req, err := stream.Recv()
+		if err == io.EOF {
+			return nil
+		}
+
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		challengeReq := req.GetChallengeRequest()
+		if challengeReq == nil {
+			return trace.BadParameter("expected PresenceMFAChallengeRequest, got %T", req)
+		}
+
+		user := actx.User.GetName()
+		u2fStorage, err := u2f.InMemoryAuthenticationStorage(actx.authServer.Identity)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		authChallenge, err := actx.authServer.mfaAuthChallenge(ctx, user, u2fStorage)
```

Comment:
I went with this approach since it's a lot more practical without bloating the ClientI interface with useless methods. It's similar to the approach used by other MFA related API's for challenge verification, hence why I went with it. I split out the parts that touch the auth server into a freestanding function and added docs on why.